### PR TITLE
Fixes #19295 - use new DHCP API

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Review by timogoebel — APPROVED** [review]
Code looks good to me, thanks @ares!

File: lib/proxy_api/dhcp.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def set(subnet, args)
     # [+mac+]    : String in coloned sextuplet format
     # Returns    : Boolean status
     def delete(subnet, mac)
-      parse super("#{subnet}/#{mac}")
+      parse super("#{subnet}/mac/#{mac}")
     rescue RestClient::ResourceNotFound
```
Comment (dLobatog) [inline lib/proxy_api/dhcp.rb]: Seems like this rescue would never be reached under normal conditions, 404s are ignored and treated like "OK, it's deleted anyway" 

https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/dhcp_api.rb#L139

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ares 👍 to me - can you double check the 404 thing? Thanks for fixing this

File: lib/proxy_api/dhcp.rb (hunk 1 above)
Comment (timogoebel) [reply lib/proxy_api/dhcp.rb]: Yes. Good catch, that did change between the old and new API.

Comment (ares) [reply lib/proxy_api/dhcp.rb]: right, I removed the rescue

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ares , @timogoebel !

**Pull request merged**